include/mata/nft/nft.hh

```diff
@@ -75,11 +74,12 @@ public:
     explicit Nft(const unsigned long num_of_states, StateSet initial_states = {},
                  StateSet final_states = {}, std::vector<Level> levels = {}, const Level levels_cnt = 1, Alphabet*
                  alphabet = nullptr)
-        : mata::nfa::Nfa(num_of_states, std::move(initial_states), std::move(final_states), alphabet), levels(std::move(levels)), levels_cnt(levels_cnt) {}
+        : mata::nfa::Nfa(num_of_states, std::move(initial_states), std::move(final_states), alphabet),
+        levels(levels.size() ? std::move(levels) : std::vector<Level>(num_of_states, 0)), levels_cnt(levels_cnt) {}
 
     explicit Nft(const mata::nfa::Nfa& other)
         : mata::nfa::Nfa(other.delta, other.initial, other.final, other.alphabet),
-          levels(std::vector<Level>(other.num_of_states(), 0)), levels_cnt(1) {}
+          levels(std::vector<Level>(other.delta.num_of_states(), 0)), levels_cnt(1) {}
```

Comment:
Why just the number of states for Delta? There can be a state which is final or initial which is not in Delta and yet needs a level. It should be `other.num_of_states()` as it was earlier, no?